tests/test_csp_rendering.py

```diff
@@ -63,70 +70,97 @@ def _fail_on_invalid_html(self, content: bytes, parser: HTMLParser):
             msg = self._formatMessage(None, "\n".join(default_msg))
             raise self.failureException(msg)
 
-    @override_settings(
-        MIDDLEWARE=settings.MIDDLEWARE + ["csp.middleware.CSPMiddleware"]
-    )
     def test_exists(self):
         """A `nonce` should exist when using the `CSPMiddleware`."""
-        response = cast(HttpResponse, self.client.get(path="/regular/basic/"))
-        self.assertEqual(response.status_code, 200)
-
-        html_root: Element = self.parser.parse(stream=response.content)
-        self._fail_on_invalid_html(content=response.content, parser=self.parser)
-        self.assertContains(response, "djDebug")
-
-        namespaces = get_namespaces(element=html_root)
-        toolbar = list(DebugToolbar._store.values())[0]
-        nonce = str(toolbar.request.csp_nonce)
-        self._fail_if_missing(
-            root=html_root, path=".//link", namespaces=namespaces, nonce=nonce
-        )
-        self._fail_if_missing(
-            root=html_root, path=".//script", namespaces=namespaces, nonce=nonce
-        )
+        for middleware in [MIDDLEWARE_CSP_BEFORE, MIDDLEWARE_CSP_LAST]:
+            with self.settings(MIDDLEWARE=middleware):
```

Comment:
It become important to test with both configurations of middleware. Or at least it will be more important if we stop relying on `request._csp_nonce`